code/modules/admin/smites/puzzgrid.dm

```diff
@@ -0,0 +1,118 @@
+/// Turns the user into a puzzgrid
+/datum/smite/puzzgrid
+	name = "Puzzgrid"
+
+	var/timer
+	var/gib_on_loss
+
+/datum/smite/puzzgrid/configure(client/user)
+	var/timer = input(user, "How long should other people have to solve the grid? 0 gives infinite time.", "Puzzgrid", 0) as num | null
```

Comment:
Any reason for this to not be tgui input ?

Reply:
tgui number input doesn't support cancelling IIRC? I could be wrong